js/src/components/google-account-card/use-google-connect-flow.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { glaData } from '.~/constants';
+import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
+import useGoogleAuthorization from '.~/hooks/useGoogleAuthorization';
+
+/**
+ * A hook that returns a connect handler that initiates Google Connect with support for incremental OAuth process.
+ *
+ * The `handleConnect` handler is meant to be used in button click handler. Upon button click, the handler will:
+ *
+ * 1. Call `fetchGoogleConnect` from `useGoogleAuthorization` hook to get the Google OAUth URL.
+ * 2. Redirect the browser to the URL.
+ * 3. If there is an error in the above process, it will display an error notice.
+ *
+ * @param {string} [loginHint] Specify the email to be requested additional scopes. Set this parameter only if wants to request a partial oauth to Google.
+ * @see https://developers.google.com/identity/protocols/oauth2/openid-connect#login-hint
+ * @return {Array} `[ handleConnect, fetchResult ]`
+ * 		- `handleConnect` is meant to be used as button click handler.
+ * 		- `fetchResult` is the same returned object from `useApiFetchCallback`.
```

Comment:
💅 
Suggest avoiding naming a data payload object starting with a verb.

Reply:
Done in [`035aceb` (#1074)](https://github.com/woocommerce/google-listings-and-ads/pull/1074/commits/035aceb35afea0ec0ad5771c9fc0ae89700187c9).